### ISSUE TITLE
field not null (edit widget photo/file name) - wrong error message if it is empty

### ISSUE THREAD

**Issue by josemvm** [3liz/lizmap-web-client#572]
if the photo field is empty, the message should be this:

![field](https://cloud.githubusercontent.com/assets/10053874/25481390/dba42232-2b44-11e7-9c9e-a1f983073b3e.jpg)

not this general:
![error](https://cloud.githubusercontent.com/assets/10053874/25481509/656deb1a-2b45-11e7-9958-8c7813738030.jpg)

thanks

**Comment by josemvm** [comment]
some behavior for date widget

**Comment by rldhont** [comment]
@josemvm can you test https://github.com/3liz/lizmap-web-client/pull/1111 or actual release_3_2 branch ?

**Comment by josemvm** [comment]
hi @rldhont , the problem remains

**Comment by laurentj** [comment]
I tried to fix the current code, but it is complicated to modify it to fix this bug, because of some limitation of jForms (the form component of Jelix). It is better to encapsulate all the logic of the upload field into a jForms plugin, to benefit of the feature of jForms, and so to avoid kind of bugs like this.

Hopefully, a plugin implementing the logic of the current code of upload, has been added to Jelix recently. We should use it and replace all "home made" code for upload. But I prefer to do this replacement into a non-stable version (3.3), to avoid regression in the 3.2 version.
